rust/operator-binary/src/restart_controller/pod.rs

```diff
@@ -192,6 +190,48 @@ async fn reconcile(pod: Arc<PartialObjectMeta<Pod>>, ctx: Arc<Ctx>) -> Result<Ac
     }
 }
 
+/// Reports the result of reconciliation.
+///
+/// The Pod restart controller has special handling, as it produced lot's of error messages below.
+/// They are expected, as we intentionally use the `Evict` API to restart Pods before e.g. the
+/// certificate expires. We roll out PDBs by default. If we try to restart multiple Pods that are
+/// part of a PDB, we get this errors.
+/// Because of this, we don't emit an error for this case, but only product a INFO trace.
+///
+/// `ERROR stackable_operator::logging::controller: Failed to reconcile object controller.name="pod.restarter.commons.stackable.tech" error=reconciler for object Pod.v1./trino-worker-default-0.default failed error.sources=[failed to evict Pod, ApiError: Cannot evict pod as it would violate the pod's disruption budget.: TooManyRequests (ErrorResponse { status: "Failure", message: "Cannot evict pod as it would violate the pod's disruption budget.", reason: "TooManyRequests", code: 429 }), Cannot evict pod as it would violate the pod's disruption budget.: TooManyRequests]`
+#[allow(clippy::type_complexity)] // The result type complexity comes from kube-rs and is what it is
+async fn report_result(
+    result: Result<
+        (ObjectRef<PartialObjectMeta<Pod>>, Action),
+        controller::Error<Error, watcher::Error>,
+    >,
+    event_recorder: Arc<Recorder>,
+) {
+    if let Err(controller::Error::ReconcilerFailed(
+        Error::EvictPod {
+            source: evict_pod_error,
+        },
+        pod,
+    )) = &result
+    {
+        const TOO_MANY_REQUESTS_HTTP_CODE: u16 = StatusCode::TOO_MANY_REQUESTS.as_u16();
+        if let kube::Error::Api(ErrorResponse {
+            code: TOO_MANY_REQUESTS_HTTP_CODE,
+            ..
+        }) = evict_pod_error
+        {
+            tracing::info!(
+                %pod,
```

Comment:
We should use semantic conventions here: `k8s.pod.name` (from memory).

Reply:
The problem here is that `pod` is a `ObjectRef<DynamicObject>`, it currently renders as `Pod.v1./simple-zk-server-default-17.default`.
We could parse name and namespace separately out of it, but that would be fallible (a DynamicObject does not need to have a namespace) and would require more error handling. Do you think we should do that? :)

Reply:
> we cannot assume a to be a Pod. It can be anything.

The compiler can't. Be we know the DynamicObject will always be a Pod. The likelihood of this changing is very very low I'd say.
Happy to go with `k8s.object_ref`, adopted in d43f3d1

Reply:
I'd say we should use `k8s.object.ref` instead of `k8s.object_ref`.

Reply:
> Be we know the DynamicObject will always be a Pod

We know "at this very moment" is not the same as "this can only be used for pods".

But yeah, happy with the replacement (but insist on the `.` and not `_` (so we base it on how upstream does it for everything else, eg `k8s.pod.name` instead of `k8s.pod_name`).